rust/lancedb/src/polars_arrow_convertors.rs

```diff
@@ -0,0 +1,123 @@
+/// Polars and LanceDB both use Arrow for their in memory-representation, but use
```

Comment:
polars wildcards their arrow-rs dependencies, so we should be able to use their conversion traits without worrying about compatibility issues

```toml
# polars/Cargo.toml
arrow-array = { version = ">=41", default-features = false }
arrow-buffer = { version = ">=41", default-features = false }
arrow-data = { version = ">=41", default-features = false }
arrow-schema = { version = ">=41", default-features = false }
```

Reply:
Please see the discussion here: https://github.com/lancedb/lancedb/pull/1235. The polars-arrow conversion traits pulled in arrow-rs 51, but LanceDB depended on arrow-rs 50. Dependency resolution did converge on version 50 and pulled in both versions. This prevented compilation.

It looks like LanceDB just upgraded to arrow-rs 51, so for now, the polars-arrow conversion traits should compile fine but @westonpace mentioned that we should assume that LanceDB will lag behind the newest version of arrow-rs.

Let me know if you think there's a way to use the polars-arrow conversion traits given this.

Reply:
> polars wildcards their arrow-rs dependencies, so we should be able to use their conversion traits without worrying about compatibility issues

That was my original thinking too but https://doc.rust-lang.org/cargo/reference/resolver.html#unexpected-dependency-duplication seems to suggest otherwise :(